aws/resource_aws_api_gateway_vpc_link.go

```diff
@@ -188,13 +195,13 @@ func apigatewayVpcLinkRefreshStatusFunc(conn *apigateway.APIGateway, vl string)
 	}
 }
 
-func waitForApiGatewayVpcLinkDeletion(conn *apigateway.APIGateway, vpcLinkID string) error {
+func waitForApiGatewayVpcLinkDeletion(conn *apigateway.APIGateway, vpcLinkID string, timeout time.Duration) error {
 	stateConf := resource.StateChangeConf{
 		Pending: []string{apigateway.VpcLinkStatusPending,
 			apigateway.VpcLinkStatusAvailable,
 			apigateway.VpcLinkStatusDeleting},
 		Target:     []string{""},
-		Timeout:    5 * time.Minute,
```

Comment:
Hi @philippevidal80 👋  Thank you for this contribution.

This is situation where increasing the encoded timeout to match the API behavior feels more appropriate, rather than introducing new configuration, since the API operation does not have any operator managed scaling factor (such as data size, etc.). Please see the [issue comments](https://github.com/terraform-providers/terraform-provider-aws/issues/10405#issuecomment-667242303) for more context.

Can you please revert all the changes and just increase the timeouts to 20 (or 30) minutes? We can get this merged immediately afterwards. Thanks!